test_util/test_upgrade_vpc.py

```diff
@@ -173,9 +172,19 @@ def create_marathon_dns_app(app_id: str, healthcheck_app_id: str) -> dict:
     # DNS resolution app to make sure DNS is available during the upgrade.
     # Periodically resolves the healthcheck app's domain name and logs whether
     # it succeeded to a file in the Mesos sandbox.
+
+    dns_parsed_app_name = marathon_app_id_to_mesos_dns_subdomain(healthcheck_app_id)
+    resolve_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(dns_parsed_app_name)
+
     return {
         "id": '/' + app_id,
         "cmd": """
+until host -W $TIMEOUT_SECONDS $RESOLVE_NAME
+do
+    echo "Waiting to resolve $RESOLVE_NAME for the first time..."
+    sleep 1
+done
```

Comment:
Why is this change necessary? Seems unrelated to changing the name that this test resolves.

If `$RESOLVE_NAME` can never be resolved, this app will produce an empty log, which `test_app_dns_survive_upgrade()` will interpret as success (no resolution failures in an empty list). If we go this route, we'll need to add an assertion that the log has at least one entry.